forwarder/src/lib.rs

```diff
@@ -137,6 +140,49 @@ impl AuroraForwarder {
             ft_transfer_call
         }
     }
+
+    /// Callback for depositing wNEAR.
+    ///
+    /// # Panics
+    ///
+    /// The callback panics if the use has a low balance.
+    #[must_use]
+    pub fn deposit_wrap_near(&self, token_id: &AccountId) -> U128 {
+        assert_self();
+        let amount = env::account_balance()
+            .checked_sub(MINIMUM_BALANCE)
+            .filter(|a| *a > 0)
+            .expect("Too low balance");
+
+        ext_wnear::ext(token_id.clone())
+            .with_attached_deposit(amount)
+            .with_static_gas(NEAR_DEPOSIT_GAS)
+            .near_deposit();
+
+        amount.into()
+    }
+
+    fn forward_nep141_token(token_id: &AccountId) -> Promise {
+        ext_token::ext(token_id.clone())
+            .ft_balance_of(env::current_account_id())
+            .then(
+                Self::ext(env::current_account_id())
+                    .with_attached_deposit(env::attached_deposit())
+                    .with_static_gas(CALCULATE_FEES_CALLBACK_GAS)
+                    .calculate_fees_callback(token_id),
+            )
+    }
+
+    fn forward_native_token(wnear_contract_id: &AccountId) -> Promise {
+        Self::ext(env::current_account_id())
+            .deposit_wrap_near(wnear_contract_id)
```

Comment:
Why bother doing this as a cross-contract call instead of calling `deposit_wrap_near` directly?

Reply:
Because we need an `amount` as a result of the promise for the `calculate_fees_callback`, and the `near_deposit` returns nothing. To be honest I couldn't get how to do it in another way.

Reply:
I would refactor with something like https://github.com/aurora-is-near/aurora-forwarder-contractrs/commit/eefdf4cb360b9abc1816438733890f1a33408b70

Reply:
Looks like just refactoring, or are we getting rid of additional promise in such a way?

Reply:
It removes the first self-call, but keeps the fees and finalizing promise calls.

Reply:
Thanks 🙏